src/api/v1/files.py

```diff
@@ -413,3 +420,96 @@ def get_latest_file_chat(
         title=chat.title if chat else None,
         history=chat.get_chat_history() if chat else [],
     )
+
+
+@router.post("/{file_id}/sql/query")
+@require_access(allowed_roles=[Role.VIEWER, Role.EDITOR, Role.OWNER])
+def run_sql_query(
+    file_id: int,
+    request: schemas.SQLQueryRequest,
+    db: Session = Depends(get_db_connection),
+    current_user: schemas.User = Depends(get_current_active_user),
+) -> schemas.SQLQueryResponse:
+    file = get_file_from_db(db, file_id)
+    MAX_RESULT_LIMIT = 10000
+
+    if not duckdb_utils.is_sql_file(file.magic_text):
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="File is not a supported SQL format.",
+        )
+
+    # Check if the query contains a LIMIT clause
+    if "limit" not in request.query.lower() and "sample" not in request.query.lower():
```

Comment:
If you already use MAX_RESULT_LIMIT why not add a "limit" statement to the SQL statement if it does not contain it (and print a warning or something)?

Reply:
makes sense.